src/code.l

```diff
@@ -1102,13 +1102,15 @@ NUMBER {INTEGER_NUMBER}|{FLOAT_NUMBER}
                                           yyextra->name+=yytext;
                                         }
 <Body>{SCOPENAME}/{BN}*[:;,)\]]         { // "int var;" or "var, var2" or "debug(f) macro" , or int var : 5;
+                                          if (QCString(yytext).startsWith("typedef")) REJECT;
```

Comment:
Shouldn't we use something like 

```
if (QCString(yytext)=="typedef" || QCString(yytext).startsWith("typedef "))
``` 
here otherwise a variable called `typedefinition` will also be matched?

Reply:
I oversaw this corner case, though I see that it might be that also "typedef:" (for e.g `typedef::my_type T5;`) has to be supported. There might be even more characters have to be supported, I'm not sure which characters would be possible though.

Please adjust as would be best.